src/classic/clvm_tools/stages/stage_2/optimize.rs

```diff
@@ -436,15 +462,22 @@ fn children_optimizer(
                 _ => {}
             }
 
-            m! {
-                optimized <- mapM(
-                    allocator,
-                    &mut list.into_iter(),
-                    &|allocator, v| {
-                        optimize_sexp(allocator, v, eval_f.clone())
-                    }
-                );
+            let mut optimized = Vec::new();
+            for item in list.iter() {
+                let res = optimize_sexp_(allocator, memo.clone(), *item, eval_f.clone())?;
+                if !equal_to(allocator, *item, res) {
```

Comment:
```suggestion
                if !different && !equal_to(allocator, *item, res) {
```

This would probably be an extra optimization.

Reply:
I still think this change would be an improvement. or is `equal_to()` so cheap?

Reply:
i made it cheaper, let me see how that works out